src/ssl.c

```diff
@@ -12005,6 +12005,166 @@ static int wolfSSL_remove_ciphers(char* list, int sz, const char* toRemove)
 
     return totalSz;
 }
+/*                                                                          */
+/* build enabled cipher list w/ TLS13 or w/o TLS13 suites                   */
+/* @param ctx    a pointer to WOLFSSL_CTX structure                         */
+/* @param suites currently enabled suites                                   */
+/* @param onlytlsv13suites flag whether correcting w/ TLS13 suites          */
+/*                         or w/o TLS13 suties                              */
+/* @param list   suites list that user wants to update                      */
+/* @return suites list on successs, otherwise NULL                          */
+static char* buildEnabledCipherList(WOLFSSL_CTX* ctx, Suites* suites,
+           int onlytlsv13suites, const char* list)
+{
+    word32 idx = 0;
+    word32 listsz = 0;
+    word32 len = 0;
+    word32 ianasz = 0;
+    const char* enabledcs = NULL;
+    char* locallist = NULL;
+    char* head = NULL;
+    byte cipherSuite0;
+    byte cipherSuite;
+    
+    /* sanity check */
+    if (ctx == NULL || suites == NULL || list == NULL)
+        return NULL;
+        
+    if (!suites->setSuites)
+        return NULL;
+    
+    listsz = (word32)XSTRLEN(list);
+    
+    /* calculate necessary buffer length */
+    for(idx = 0; idx < suites->suiteSz; idx++) {
+        
+        cipherSuite0 = suites->suites[idx];
+        cipherSuite  = suites->suites[++idx];
+        
+        if (onlytlsv13suites && cipherSuite0 == TLS13_BYTE) {
+            enabledcs = GetCipherNameInternal(cipherSuite0, cipherSuite);
+        } else if (!onlytlsv13suites && cipherSuite0 != TLS13_BYTE) {
+            enabledcs = GetCipherNameInternal(cipherSuite0, cipherSuite);
+        } else
+            continue;
+        
+        if (XSTRNCMP(enabledcs, "None", XSTRLEN(enabledcs)) != 0) {
+            len += (word32)XSTRLEN(enabledcs) + 2;
+        }
+    }
+    
+    len += listsz + 2;
+    
+    /* build string */
+    if (len > 0) {
```

Comment:
Can this fail given statement above?
Change to compare with 2 I assume.

Also return NULL when nothing to return and then no need to put the construction code in a block.

Reply:
Thanks. Changed to compare with 2.